src/python/pants/backend/java/dependency_inference/rules.py

```diff
@@ -68,13 +68,13 @@ async def infer_java_dependencies_via_imports(
         types.update(dependency_name(imp.name, imp.is_static) for imp in analysis.imports)
     if java_infer_subsystem.consumed_types:
         package = analysis.declared_package
-        if package:
-            types.update(
-                f"{package}.{consumed_type}"
-                for consumed_type in analysis.consumed_unqualified_types
-            )
-        else:
-            types.update(analysis.consumed_unqualified_types)
+
+        maybe_qualify_types = (
```

Comment:
Maybe add a comment to explain this logic? I introduced the bug and am still confused.

Reply:
Sure thing. 

To close the loop here: I was seeing `javax.inject.Inject` get included in `consumed_unqualified_types` as `com.google.inject.javax.inject.Inject`, which is not correct.